src/Jackalope/Version/VersionHistory.php

```diff
@@ -166,9 +174,11 @@ public function getVersion($versionName)
      */
     public function getVersionByLabel($label)
     {
-        // @codeCoverageIgnoreStart
-        throw new NotImplementedException();
-        // @codeCoverageIgnoreEnd
+        if ($this->hasVersionLabel($label)) {
+            return $this->versionLabels[$label];
+        } else {
```

Comment:
no need for an else, we already returned above. just throwing the exception is enough. or maybe invert, if (!$this->hasVersionLabel...) throw. then return the version.